src/main/java/gregtech/api/gui/widgets/SituationWidget.java

```diff
@@ -0,0 +1,100 @@
+package gregtech.api.gui.widgets;
+
+import gregtech.api.Situation;
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.IRenderContext;
+import gregtech.api.gui.Widget;
+import gregtech.api.gui.resources.TextureArea;
+import gregtech.api.util.Position;
+import gregtech.api.util.Size;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.item.ItemStack;
+import net.minecraft.network.PacketBuffer;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import java.util.Collections;
+import java.util.function.IntSupplier;
+
+import static gregtech.api.Situation.*;
+import static gregtech.api.Situations.SituationsTypes.*;
+
+public class SituationWidget extends Widget {
+
+    private final IntSupplier currentSituationId;
+    protected String tooltipHoverString;
+    protected long hoverStartTime = -1L;
+    protected boolean isMouseHovered;
+    protected int currentError;
+
+    protected TextureArea area;
+    private boolean isVisible = true;
+
+    public SituationWidget(int xPosition, int yPosition, int width, int height, IntSupplier getSituationId) {
```

Comment:
Also there is still lingering my previous comment regarding use `IntSuplier` here

Reply:
i couldn't find any other way to sync the widget properly while its open without using it here.
setting it to int will only display the changes when the GUI is reopened.

Reply:
For synchronization you will still need that int id but you can get it from Situation it self